x-pack/libbeat/autodiscover/providers/aws/elb/config.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package elb
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/autodiscover/template"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Config for the aws_elb autodiscover provider.
+type Config struct {
+	Type string `config:"type"`
+
+	// Standard autodiscover fields.
+
+	// Hints are currently not supported, but may be implemented in a later release
+	HintsEnabled bool                    `config:"hints.enabled"`
+	Builders     []*common.Config        `config:"builders"`
+	Appenders    []*common.Config        `config:"appenders"`
+	Templates    template.MapperSettings `config:"templates"`
+
+	// Period defines how often to poll the AWS API.
+	Period time.Duration `config:"period" validate:"nonzero,required"`
+
+	// AWS Specific autodiscover fields
+
+	Regions         []string `config:"regions" validate:"required"`
```

Comment:
In aws metricbeat module, we allow user to not set `Regions`. In that case, we will go query for all regions and loop through them. This might not be a good practice here but I just want to point it out.

Reply:
Thanks for the heads up. Normally I'd say we should not change behavior, however, maybe we should make this a requirement going forward. There are a lot of AWS regions, and it seems like a waste. Not to mention, the user may only have perms for one region. Looping through them all may create unnecessary errors.

WDYT?

Reply:
Yeah good point on the permission problem!